### PR TITLE
ci: break CI actions into smaller pieces

### PR DESCRIPTION
**Pull request by kpodsiad**
I removed `onload` resources generation and then I extracted two additional actions on CI. 
Finer grained CI actions should reduce main tests duration and provide a better experience when debugging CI.

### CODE REVIEW

**Comment by kpodsiad** [comment]
Before
Screenshot 2022-03-28 at 15 24 12
After
Screenshot 2022-03-28 at 15 24 18

Although these are statistics from a single run they indicate that CI jobs on GraalVM are a bit faster. The most gain can be observed on the slowest windows so it's a double win.

File: bin/setup-test-projects.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (tgodzik) [inline bin/setup-test-projects.sh:1]: Do we need this bash script if the same can be achieved by exportProjectsInTestResources?

File: docs/contributing-guide.md [hunk 2]
```diff
@@ -39,8 +39,10 @@ bsp/cli connection to it.
 1. Clone the [GitHub repository](https://github.com/scalacenter/bloop)
 1. Change directory to bloop's base directory and run `git submodule update --init`.
 1. Run the sbt shell in the base directory.
+1. If it's the first time you should run `exportProjectsInTestResources` command to generate test projects which are needed for tests.
```
Comment (tgodzik) [inline docs/contributing-guide.md:42]: Do we show a sensible exception if the project is not present?

**Review by tgodzik — COMMENTED** [review]
The approach looks sensible, thanks for looking into that.

**Comment by dos65** [comment]
@kpodsiad I did an alternative fix for this in #1708

**Comment by tgodzik** [comment]
@kpodsiad looks like we can close this one?

**Comment by kpodsiad** [comment]
Actually, after https://github.com/scalacenter/bloop/pull/1708 was merged I pivoted into breaking CI actions into smaller pieces. I think it will be valuable to have more fine-grained actions which are responsible only for testing something more narrow than currently.

**Comment by kpodsiad** [comment]
@tgodzik wdyt?

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**